### ISSUE TITLE
False positive Style/CaseEquality for range checks

### ISSUE THREAD

**Issue by obfuscoder** [bbatsov/rubocop#2443]
I have the following code fragment:

```ruby
range = 1..10
value = 5
puts 'Within range' if range === value
```

Running rubocop on this results in this:

```
t.rb:3:30: C: Avoid the use of the case equality operator ===.
puts 'Within range' if range === value
                             ^^^
```

The === is used for checking whether value is within the range. This does not have to do with case equality checks.

**Comment by mikegee** [comment]
"case equality operator" is just what a lot of people call `===`. The top answer of this question has a great explanation of it: http://stackoverflow.com/questions/4467538/what-does-the-operator-do-in-ruby

It includes this: 

> it is usually called the triple equals operator or threequals operator or case equality operator. I really dislike those names, because this operator has absolutely nothing whatsoever to do with equality.

**Comment by scottmatthewman** [comment]
The case equality operator is defined on `range` so you can do something like:

```ruby
case my_value
when 1..100
  ...
```

The case method uses the `===` method on each supplied in the successive `when` statements.

Outside of a case statement, it's more expressive to use `include?` or `cover?` when checking to see if a value is included within a range. Thus, rubocop is giving the right advice.

**Comment by bbatsov** [comment]
Indeed.
